### PR TITLE
[SE-0302] Implement structural conformances to Sendable.

### PR DESCRIPTION
**Pull request by DougGregor**
Add implicit conformance of structural types to Sendable as appropriate:
* A tuple type is Sendable when its element types are Sendable
* A metatype type is Sendable
* A function type is Sendable if it is @sendable, thin, or C
* A builtin type is always Sendable

Implements rdar://76836578.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

File: lib/AST/Module.cpp [hunk 1]
```diff
@@ -988,6 +988,115 @@ ProtocolConformanceRef ModuleDecl::lookupConformance(Type type,
       getASTContext().evaluator, request, ProtocolConformanceRef::forInvalid());
 }
 
+/// Synthesize a builtin tuple type conformance to the given protocol, if
+/// appropriate.
+static ProtocolConformanceRef getBuiltinTupleTypeConformance(
+    Type type, const TupleType *tupleType, ProtocolDecl *protocol) {
+  // Tuple type are Sendable when all of their element types are Sendable.
```
Comment (ktoso) [inline lib/AST/Module.cpp:995]: Wohoo, that unlocks Void conforming to Sendable as well right? AFAIR at least it was related to tuples hm.

File: lib/Serialization/ModuleFormat.h [hunk 2]
```diff
@@ -2,7 +2,7 @@
 //
 // This source file is part of the Swift.org open source project
 //
-// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
```
Comment (ktoso) [inline lib/Serialization/ModuleFormat.h:5]: (minor, if you care -- technically 2021, if you're using some script to update maybe change in it?)

**Review by ktoso — APPROVED** [review]

**Review by airspeedswift — APPROVED** [review]

File: lib/AST/Module.cpp (hunk 1 above)
Comment (DougGregor) [reply lib/AST/Module.cpp:995]: Yes, that's why I'm pulling it in.

**Pull request merged**